pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+										Service: &networkingv1.IngressServiceBackend{
+											Name: comp.GetName() + "-collabora-code",
+											Port: networkingv1.ServiceBackendPort{
+												Number: 9980,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			TLS: []networkingv1.IngressTLS{
+				{
+					Hosts:      []string{comp.Spec.Parameters.Service.Collabora.FQDN},
+					SecretName: comp.GetName() + "-collabora-code-tls",
+				},
+			},
+		},
+	}
+
+	if svc.Config.Data["ingress_annotations"] != "" {
+		annotations := map[string]string{}
+
+		err := yaml.Unmarshal([]byte(svc.Config.Data["ingress_annotations"]), annotations)
+		if err != nil {
+			return fmt.Errorf("cannot unmarshal ingress annotations from input: %w", err)
+		}
+
+		if val, ok := annotations["cert-manager.io/cluster-issuer"]; ok {
+			ingress.Annotations["cert-manager.io/cluster-issuer"] = val
+		}
+	}
+
+	return svc.SetDesiredKubeObject(ingress, comp.GetName()+"-collabora-code-ingress", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createIssuer(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	issuer := &cmv1.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-issuer",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Spec: cmv1.IssuerSpec{
+			IssuerConfig: cmv1.IssuerConfig{
+				SelfSigned: &cmv1.SelfSignedIssuer{},
+			},
+		},
+	}
+
+	return svc.SetDesiredKubeObject(issuer, comp.GetName()+"-collabora-code-issuer", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createCertificate(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	certificate := &cmv1.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-certificate",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Spec: cmv1.CertificateSpec{
+			SecretName: "collabora-code-tls",
+			DNSNames:   []string{comp.Spec.Parameters.Service.Collabora.FQDN},
+			IssuerRef: certmgrv1.ObjectReference{
+				Name: comp.GetName() + "-collabora-code-issuer",
+			},
+		},
+	}
+
+	return svc.SetDesiredKubeObject(certificate, comp.GetName()+"-collabora-code-certificate", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createCoolWSDConfigMap(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "collabora-code-coolwsd-config",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Data: map[string]string{
+			"coolwsd.xml": coolwsd,
+			"coolkit.xml": coolkit,
+		},
+	}
+
+	return svc.SetDesiredKubeObject(cm, comp.GetName()+"-collabora-code-coolwsd-config", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createSecretWithRSAKeys(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	objName := comp.GetName() + "-collabora-code-ssh-key"
+
+	currentSecret := &corev1.Secret{}
+
+	err := svc.GetObservedKubeObject(currentSecret, objName)
+	if err == nil {
```

Comment:
I'd do a more granular error check here. There are other things that can go wrong other than the object not existing.

Please check explicitly for `runtime.ErrNotFound`.

Reply:
done